File: dsa-learning/dsa-python-code/com/dsa/leetcode/labuladong/array/two_pointers/P167_TwoSum2InputArrayIsSorted.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1
        while left < right:
            temp = numbers[left] + numbers[right]
            if temp > target:
                right -= 1
            elif temp < target:
                left += 1
            else:
                return [left + 1, right + 1]
        return None

File: dsa-learning/dsa-java-code/src/main/java/com/dsa/leetcode/labuladong/array/P167_TwoSum2InputArrayIsSorted.java
package com.dsa.leetcode.labuladong.array;

/**
 * @since 2023/11/4 20:41
 */
public class P167_TwoSum2InputArrayIsSorted {

    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;
        int temp;
        while (left < right) {
            temp = numbers[left] + numbers[right];
            if (temp == target) {
                return new int[]{left + 1, right + 1};
            } else if (temp > target) {
                right--;
            } else {
                left++;
            }
        }
        return null;
    }
}
